### PR TITLE
fix: Set full height for set map dialog, added help text

### PR DESCRIPTION
**Pull request by josebui**
## Description

Set full height for set map dialog, added help text

### Checklist
- [x] Corresponding issue has been opened
- [ ] New tests added

### Related Issues
Fixes https://github.com/techmatters/terraso-product/issues/231

### Verification steps

Screenshot 2023-08-15 at 12 24 29 PM
Screenshot 2023-08-15 at 12 24 38 PM

### CODE REVIEW

File: src/storyMap/components/StoryMapForm/MapLocationDialog.js [hunk 1]
```diff
@@ -18,20 +18,98 @@ import React, { useCallback, useEffect, useMemo, useState } from 'react';
...
+        </Trans>
+      </Box>
+      <Box>
+        <Trans i18nKey="storyMap.form_location_helper_text_step_3">
+          <Typography gutterBottom variant="h3">
+            Title
+          </Typography>
+          <Typography gutterBottom>Paragraph 1</Typography>
+          <img
+            src="/storyMap/set-map-step-3.png"
```
Comment (paulschreiber) [inline src/storyMap/components/StoryMapForm/MapLocationDialog.js:104]: issue: this image has text in it and needs to be localized.

File: src/localization/locales/en-US.json [hunk 2]
```diff
@@ -971,7 +971,16 @@
         "iframe_title": "Contact Terraso",
         "embed_header": "View Story Map on",
         "upload_rejected_file-too-large": "{{rejectedFiles}} cannot be added because the file is too large.",
-        "upload_rejected": "The file(s) cannot be added."
+        "upload_rejected": "The file(s) cannot be added.",
+        "form_location_dialog_helper_text_label": "How to set the map location",
+        "form_location_helper_text_step_1": "<0>Step 1. Find a location</0><1>Search for a location or find one by panning or zooming on the map.</1><2/>",
+        "form_location_helper_text_step_1_image_alt": "A screenshot of the Guyana map in the Set Map Location screen.The location search field is filled with the keyword “Guyana”.",
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:977]:
```suggestion
        "form_location_helper_text_step_1_image_alt": "A screenshot of the Guyana map in the Set Map Location screen. The location search field is filled with the keyword “Guyana”.",
```

File: src/localization/locales/en-US.json [hunk 3]
```diff
@@ -971,7 +971,16 @@
...
+        "form_location_helper_text_step_2": "<0>Step 2. Adjust the background map</0><1>Choose a location, then adjust  the map view by panning or zooming.</1><2>Select the <1 /> button to rotate the map.  Use the <3 /> button to reset the map. You can also right-click on the  mouse and drag left and right to rotate the view.</2><3 /><4 />",
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:978]:
```suggestion
        "form_location_helper_text_step_2": "<0>Step 2. Adjust the background map</0><1>Choose a location,then adjust the map view by panning or zooming.</1><2>Select the <1 /> button to rotate the map. Use the <3 /> button to reset the map. You can also right-click on the mouse and drag left and right to rotate the view.</2><3 /><4 />",
```

**Review by paulschreiber — CHANGES_REQUESTED** [review]

File: src/localization/locales/en-US.json [hunk 4]
```diff
@@ -971,7 +971,16 @@
         "iframe_title": "Contact Terraso",
         "embed_header": "View Story Map on",
         "upload_rejected_file-too-large": "{{rejectedFiles}} cannot be added because the file is too large.",
-        "upload_rejected": "The file(s) cannot be added."
+        "upload_rejected": "The file(s) cannot be added.",
```
[the remark this answers is not in the document]
Comment (josebui) [reply src/localization/locales/en-US.json]: It is just one file, I will update this to always show the message as only one error file

File: src/localization/locales/en-US.json [hunk 5]
```diff
@@ -971,7 +971,17 @@
...
+        "form_location_helper_text_step_2_1_image_alt": "A map with the rotation button pointing North and South.",
+        "form_location_helper_text_step_2_2_image_alt": "A map shows the same location as the previous map with the view rotated 90 degrees. The rotation button points east and west.",
+        "form_location_helper_text_step_3": "<0>Step 3. Set the location with desired background map</0><1>Once you are satisfied with the background map, select Set Location. Each chapter can have a different background map.</1><2 />",
+        "form_location_helper_text_step_3_image_src": "/storyMap/set-map-step-3-en.png",
+        "form_location_helper_text_step_3_image_alt": "A screenshot of the Set Map Location screen. The “Set Location” button in the top right is circled.",
+        "form_location_helper_text_bearing_icon_alt": "Bearing button icon"
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:984]:
```suggestion
        "form_location_helper_text_bearing_icon_alt": "Bearing"
```

Comment (paulschreiber) [reply src/localization/locales/en-US.json:984]: Avoids this being read as "bearing button icon button"

File: src/localization/locales/es-ES.json [hunk 6]
```diff
@@ -978,7 +978,17 @@
...
+        "form_location_helper_text_step_2_1_image_alt": "Un mapa con el botón de rotación apuntando al norte y al sur.",
+        "form_location_helper_text_step_2_2_image_alt": "Un mapa muestra la misma ubicación que el mapa anterior con la vista girada 90 grados. El botón de rotación apunta al este y al oeste.",
+        "form_location_helper_text_step_3": "<0>Paso 3. Establece la ubicación con el mapa de fondo deseado</0><1>Una vez que estés satisfecho con el mapa de fondo, selecciona Establecer ubicación. Cada capítulo puede tener un mapa de fondo diferente.</1><2 />",
+        "form_location_helper_text_step_3_image_src": "/storyMap/set-map-step-3-es.png",
+        "form_location_helper_text_step_3_image_alt": "Una captura de pantalla de la pantalla Establecer ubicación en el mapa. El botón “Establecer ubicación” en la parte superior derecha tiene un círculo.",
+        "form_location_helper_text_bearing_icon_alt": "Icono del botón de rumbo"
```
Comment (paulschreiber) [inline src/localization/locales/es-ES.json:991]:
```suggestion
        "form_location_helper_text_bearing_icon_alt": "Rumbo"
```

File: src/common/components/HelperText.js [hunk 7]
```diff
@@ -132,14 +135,26 @@ const HelperText = props => {
...
     <>
-      <IconButton
-        size="small"
-        ref={anchorEl}
-        aria-label={t('form.helper_text_info_label', { label })}
-        onClick={handleClick}
-      >
-        <InfoIcon fontSize="1rem" />
-      </IconButton>
+      {showLabel ? (
```
Comment (paulschreiber) [inline src/common/components/HelperText.js:138]: question: when is `showLabel` true?

Comment (josebui) [reply src/common/components/HelperText.js:138]: @paulschreiber It is true for this change, so in this case we need to show not just the helper text icon but also a label like this:
Screenshot 2023-08-28 at 3 37 31 PM
For most cases it is just an icon button, like this:
Screenshot 2023-08-28 at 3 38 05 PM

I think it looks similar but in the first case you can also click on the text to open the helper text, and the text is specifically for the helper button

File: src/storyMap/components/StoryMapForm/MapLocationDialog.js [hunk 8]
```diff
@@ -132,22 +203,39 @@ const MapLocationDialog = props => {
...
+                prefix
+                <i>italic</i>
+              </Trans>
+            ) : (
+              <>{t('storyMap.form_location_dialog_title_blank')}</>
+            )}
+          </DialogTitle>
+          <DialogContent sx={{ pb: 0 }}>
+            <HelperText
+              showLabel
```
Comment (josebui) [inline src/storyMap/components/StoryMapForm/MapLocationDialog.js:228]: @paulschreiber in here we are passing the true value

**Review by paulschreiber — APPROVED** [review]

**Pull request merged**